typescript/sdk/src/deploy/EvmModuleDeployer.ts

```diff
@@ -259,11 +259,20 @@ export class EvmModuleDeployer<Factories extends HyperlaneFactories> {
         `Deploying new ${threshold} of ${values.length} address set to ${chain}`,
       );
       const overrides = multiProvider.getTransactionOverrides(chain);
-      const hash = await factory['deploy(address[],uint8)'](
+
+      // estimate gas
+      const estimatedGas = await factory.estimateGas['deploy(address[],uint8)'](
         values,
         threshold,
         overrides,
       );
+
+      // add 10% buffer
+      const hash = await factory['deploy(address[],uint8)'](values, threshold, {
+        ...overrides,
+        gasLimit: estimatedGas.add(estimatedGas.div(10)), // 10% buffer
```

Comment:
Would `.mul(1.1)` work? Or maybe not because this is an Ethers BigNumber which doesn't support decimals? In which case, what happens if gas isn't divisible by 10?

Reply:
first time round i did try .mul(1.1) for simplicity but I ran into issues - I haven't thought about the not-divisible-by-10 problem before but we've also not run into it in the 4months we've had it in the multiprovider 🤷‍♂️